pkg/proxy/mark.go

```diff
@@ -14,32 +14,54 @@
 
 package proxy
 
-// Magic markers are attached to each packet. The lower 16 bits are used to
-// identify packets which have gone through the proxy and to determine whether
-// the packet is coming from a proxy at ingress or egress, or from the host.
-// The marking is compatible with Kubernetes's use of the packet mark.  The
-// upper 16 bits can be used to carry the security identity.
+// The skb mark is used to transmit both identity and special markers to
+// identify traffic from and to proxies. The mark field is being used in the
+// following way:
+//
+//  1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2
```

Comment:
It's odd to me to see this not starting from `0`

Reply:
It's bit 1

Reply:
It's a bit weird that the highest bit is the bit 1, except if thinking about storage order in network/big endian byte order. I'd find it a `bit` more natural to designate the LSB as the 1st bit.